modules/react/button/lib/parts/ButtonLabelIcon.tsx

```diff
@@ -38,52 +37,29 @@ const iconSizes: Record<ButtonSizes, number> = {
   large: 24,
 };
 
-const getIconPositionStyles = ({iconPosition, size}: ButtonLabelIconProps) => {
-  // Adjust margin values for visual balance, large buttons require 8px
-  const balancingMargin = size === 'large' ? space.xxs : space.xxxs;
-  // iconPosition can only be "right", "left", or undefined (defaults to "left")
-  if (iconPosition === 'right') {
-    return {
-      marginLeft: undefined,
-      marginRight: `-${balancingMargin} !important`,
-    };
-  } else {
-    return {
-      marginLeft: `-${balancingMargin} !important`,
-      marginRight: undefined,
-    };
-  }
-};
-
-const ButtonLabelIconStyled = styled('span', {
-  shouldForwardProp: prop => isPropValid(prop) && prop !== 'size',
-})<ButtonLabelIconProps>(
-  {
-    display: 'inline-block',
-  },
-  ({size}) => ({
-    width: size ? iconSizes[size] : iconSizes.medium,
-    height: size ? iconSizes[size] : iconSizes.medium,
-  }),
-  getIconPositionStyles
-);
-
-export const ButtonLabelIcon = ({
-  icon,
-  size,
-  iconPosition,
-  shouldMirrorIcon = false,
-  ...elemProps
-}: ButtonLabelIconProps) => {
-  if (icon === undefined) {
-    return null;
-  }
+export const ButtonLabelIcon = createComponent('span')({
+  displayName: 'ButtonLabelIconNew',
+  Component: (
+    {icon, size = 'medium', shouldMirrorIcon = false, ...elemProps}: ButtonLabelIconProps,
+    ref,
+    Element
+  ) => {
+    if (icon === undefined) {
+      return null;
+    }
 
-  const iconSize = size ? iconSizes[size] : undefined;
+    const iconSize = size ? iconSizes[size] : iconSizes.large;
```

Comment:
I'm not sure I follow the logic here. `size` has a default value of `medium`, so it should default to `iconSizes['medium']`. When would it return `iconSizes.large`, and is that what you would want? I feel like I'm missing some context.

In my head it seems like the logic should be:
```ts
const iconSize = iconSizes[size];
```